lib/pbench/test/unit/server/test_datasets_contents.py

```diff
@@ -0,0 +1,250 @@
+            try:
+                dataset_id = Dataset.query(name=dataset).resource_id
+            except DatasetNotFound:
+                dataset_id = dataset  # Allow passing deliberately bad value
+            headers = {"authorization": f"bearer {pbench_drb_token}"}
+            response = client.get(
+                f"{server_config.rest_uri}/datasets/{dataset_id}/contents/{target}",
+                headers=headers,
+            )
+            assert (
+                response.status_code == expected_status
+            ), f"Unexpected failure '{response.json}'"
+            return response
+
+        return query_api
+
+    def test_get_no_dataset(self, query_get_as):
+        response = query_get_as(
+            "nonexistent-dataset", "metadata.log", HTTPStatus.NOT_FOUND
+        )
+        assert response.json == {"message": "Dataset 'nonexistent-dataset' not found"}
+
+    def test_dataset_not_present(self, query_get_as):
+        response = query_get_as("fio_2", "metadata.log", HTTPStatus.NOT_FOUND)
+        assert response.json == {
+            "message": "The dataset tarball named 'random_md5_string4' is not found"
+        }
+
+    def test_unauthorized_access(self, query_get_as):
+        response = query_get_as("test", "metadata.log", HTTPStatus.FORBIDDEN)
+        assert response.json == {
+            "message": "User drb is not authorized to READ a resource owned by test with private access"
+        }
+
+    @pytest.mark.parametrize("key", (None, "", "subdir1"))
+    def test_path_is_directory(self, query_get_as, monkeypatch, key):
+        def mock_find_entry(_s, _d: str, path: Optional[Path]):
+            file = path if path else Path("")
+            return {
+                "children": {},
+                "details": CacheObject(
+                    file.name if key else "",
+                    file,
+                    None,
+                    None,
+                    None,
+                    CacheType.DIRECTORY,
+                ),
+            }
+
+        monkeypatch.setattr(CacheManager, "find_entry", mock_find_entry)
+        monkeypatch.setattr(Path, "is_file", lambda self: False)
+        monkeypatch.setattr(Path, "exists", lambda self: True)
+
+        response = query_get_as("fio_2", key if key else "", HTTPStatus.OK)
+        assert response.json == {
+            "directories": [],
+            "files": [],
+            "name": key if key else "",
+            "type": "DIRECTORY",
+            "uri": f"https://localhost/api/v1/datasets/random_md5_string4/contents/{key if key else ''}",
+        }
+
+    def test_not_a_file(self, query_get_as, monkeypatch):
+        def mock_find_entry(_s, _d: str, path: Optional[Path]):
+            raise BadDirpath("Nobody home")
+
+        monkeypatch.setattr(CacheManager, "find_entry", mock_find_entry)
+        monkeypatch.setattr(Path, "is_file", lambda self: False)
+        monkeypatch.setattr(Path, "exists", lambda self: False)
```

Comment:
Given that the call to `find_entry()` is going to raise an exception, do these mocks actually get called?  (Ditto lines 80 & 81.)